extending-dummy4j.md

```diff
@@ -41,20 +41,29 @@ If you choose to, you can place all of your definitions in one file. You can als
 
 ## Parser
 
-The parser recognizes two placeholders:
+The parser recognizes three placeholders:
 * `#{key.path}` - will resolve to a random value from the list of data definitions using the provided key path; The
- resolved value may itself be an expression
+ resolved value may itself be an expression. If the path resolves to a list of keys (instead of values),
+ a random one of them will be returned. 
+ The value will always be taken from a single locale (the first one which contains values for the path).
+* `#{{key.path}}` (since SNAPSHOT) - like above, but will return a random value from the superset of all locales' 
```

Comment:
The following definition file is broken in the `mixed` group when we try to resolve a random value from all locales:
```
en:
  flavour:
    chocolate: [
        "white chocolate", "dark chocolate", "chocolate"
    ]
    nutty: [
        "hazelnut", "almond", "pecan", "coconut", "nougat"
    ]
    mixed: [
        "#{flavour.chocolate} and #{flavour.nutty}",
    ]
fr:
  flavour:
    chocolate: [
        "fr"
    ]
    nutty: [
        "fr2", "fr2", "fr2"
    ]
    mixed: [
        "#{flavour.chocolate} and #{flavour.nutty}",
    ]
```
Calling `dummy.listOf(20, () -> dummy.expressionResolver().resolve("#{{flavour.mixed}}"))` will return only english flavours (probably because it's first in the file).

To solve this issue, we would have to change how `LocalizedDummyDefinitions` works and make `ExpressionResolver` remember the last resolved locale.

Reply:
This was indeed broken. I have fixed it.